pkg/model/serializer.go

```diff
@@ -724,10 +724,27 @@ func checkFileAndNotDirectory(path string) error {
 func (d *Dev) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	type devType Dev // Prevent recursion
 	dev := devType(*d)
-	err := unmarshal(&dev)
+	type hybridModeInfo struct {
+		Workdir     string            `json:"workdir,omitempty" yaml:"workdir,omitempty"`
+		Selector    Selector          `json:"selector,omitempty" yaml:"selector,omitempty"`
+		Forward     []forward.Forward `json:"forward,omitempty" yaml:"forward,omitempty"`
+		Environment Environment       `json:"environment,omitempty" yaml:"environment,omitempty"`
+		Command     Command           `json:"command,omitempty" yaml:"command,omitempty"`
+		Reverse     []Reverse         `json:"reverse,omitempty" yaml:"reverse,omitempty"`
+		Mode        string            `json:"mode,omitempty" yaml:"mode,omitempty"`
+	}
+	hybridModeDev := &hybridModeInfo{}
+	err := unmarshal(hybridModeDev)
 	if err != nil {
-		return fmt.Errorf("Unmarshal error: '%s'", err)
+		if hybridModeDev.Mode == "hybrid" {
+			return err
+		}
```

Comment:
Is it guaranteed that this will get filled? 

If not, I think we might need to do this in 2 steps, first unmarshall into a struct with only `Mode` and check there to see how to follow


---

Also, we could wrap the error here to notify the user that the hybrid mode has fewer fields

Reply:
following spec, if hybrid mode is not filled we assume that we are working with the classic mode: sync

Reply:
@AdrianPedriza agreed. I'm talking about the unmarshalling. In this block we are checking `hybridModeDev.Mode` but the marshalling has failed. Are we guaranteed to have it filled when `mode` is set?

Reply:
if the unmarshal fail because the dev is not hybrid or because it contains field that doesn't belong to hybrid mode I'm still able to check this value if user filled it in. So let's think on an scenario where user use this dev:
``````yaml

dev:
  frontend:
    mode: hybrid
    selector:
      app.kubernetes.io/part-of: okteto
      app.kubernetes.io/component: frontend
    command: ["sh", "-c", "yarn start"]
    reverse:
      - 8080:8080
    fordward:
      - 8090:8090
``````
(this will fail because `fordward` field is not supported for hybrid mode)
first of all, we will try to unmarshal it using hybrid field. It will fail but `mode` value will be filled in the struct because that value is present. If there is an error and the value is hybrid (this case) there is nothing more we can do, we return err. On the other hand, if there is an error trying to unmarshal it to hybrid mode and there is no indication that mode is hybrid then we fall back to the classic unmarshal. Is that what you are asking?

Reply:
ok, tell me if this a9e3e62cd08289de9b575f09efa40e4f7b97d70f covers your concerns :)